src/PrivateDns/PrivateDns/Zones/GetAzurePrivateDnsZone.cs

```diff
@@ -0,0 +1,52 @@
+﻿// ------------------------------------------------------------------------------------------------
+// <copyright file="GetAzurePrivateDnsZone.cs" company="Microsoft Corporation">
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// </copyright>
+// ------------------------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.PrivateDns.Zones
+{
+    using System.Management.Automation;
+    using Microsoft.Azure.Commands.PrivateDns.Models;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+    /// <summary>
+    /// Gets one or more existing zones.
+    /// </summary>
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "PrivateDnsZone", DefaultParameterSetName = "Default"), OutputType(typeof(PrivateDnsZone))]
+    public class GetAzurePrivateDnsZone : PrivateDnsBaseCmdlet
+    {
+        private const string ParameterSetResourceGroup = "ResourceGroup";
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, ParameterSetName = ParameterSetResourceGroup, HelpMessage = "The full name of the private zone (without a terminating dot).")]
```

Comment:
using ValueFromPipelineByPropertyName instead of ValueFromPipeline for resourceId causes my tests to fail. since we don't pipe resourceid by property name maybe we should use (ValueFromPipeline = true) instead?

Reply:
also to run command like: $createdZone2 = $resourceGroup | New-AzPrivateDnsZone -Name $zoneName2

we need those ValueFromPipelineByPropertyName on params like Name etc.

Reply:
changed it.